bigtable/tests/admin_integration_test.cc

```diff
@@ -12,185 +12,280 @@
+#include <vector>
 
-namespace {
 namespace admin_proto = ::google::bigtable::admin::v2;
 
-void CheckInstanceIsEmpty(bigtable::TableAdmin& admin) {
-  std::vector<std::string> table_names;
-  auto table_list = admin.ListTables(admin_proto::Table::NAME_ONLY);
-  if (not table_list.empty()) {
+namespace {
+
+class AdminIntegrationTest : public bigtable::testing::TableIntegrationTest {
+ protected:
+  std::unique_ptr<bigtable::TableAdmin> table_admin_;
+
+  void SetUp() {
+    std::shared_ptr<bigtable::AdminClient> admin_client =
+        bigtable::CreateDefaultAdminClient(
+            bigtable::testing::TableTestEnvironment::project_id(),
+            bigtable::ClientOptions());
+    table_admin_ = absl::make_unique<bigtable::TableAdmin>(
+        admin_client, bigtable::testing::TableTestEnvironment::instance_id());
+  }
+
+  void TearDown() {}
+
+  bool TestForTableListCheck(std::vector<std::string> expected_table_list) {
+    std::vector<std::string> actual_table_list;
+    std::vector<std::string> diff_table_list;
+
+    auto table_list = table_admin_->ListTables(admin_proto::Table::NAME_ONLY);
+    for (auto const& table_name : table_list) {
+      actual_table_list.push_back(table_name.name());
+    }
+
+    // Sort actual_table_list
+    sort(actual_table_list.begin(), actual_table_list.end());
+    // Sort expected_table_list
+    sort(expected_table_list.begin(), expected_table_list.end());
+
+    // Get the difference of expected_table_list and actual_table_list
+    set_difference(expected_table_list.begin(), expected_table_list.end(),
+                   actual_table_list.begin(), actual_table_list.end(),
+                   back_inserter(diff_table_list));
+
+    if (not diff_table_list.empty()) {
+      std::ostringstream ostring_stream;
+      ostring_stream << "Mismatched Tables: "
+                     << absl::StrJoin(diff_table_list, "\n")
+                     << "\nactual: " << absl::StrJoin(actual_table_list, "\n")
+                     << "\nexpected: "
+                     << absl::StrJoin(expected_table_list, "\n") << "\n";
+      std::cout << ostring_stream.str();
+    }
+
+    return diff_table_list.empty();
+  }
+
+  bool CheckTableSchema(admin_proto::Table const& actual_table,
+                        std::string const& expected_text,
+                        std::string const& message) {
+    admin_proto::Table expected_table;
+
+    if (not google::protobuf::TextFormat::ParseFromString(expected_text,
+                                                          &expected_table)) {
+      std::ostringstream ostring_stream;
+      ostring_stream << message << ": could not parse protobuf string <\n"
+                     << expected_text << ">\n";
+      std::cout << ostring_stream.str();
+
+      return false;
+    }
+
+    std::string delta;
+    google::protobuf::util::MessageDifferencer message_differencer;
+    message_differencer.ReportDifferencesToString(&delta);
+    bool is_msg_diff_notpresent =
```

Comment:
In general we prefer variables that say what they are, for example here it could be `messages_compare_equal`.  That is easier to reason about that "difference not present".